### PR TITLE
Provide common settings for HLS and Vivado compilation

### PR DESCRIPTION
**Pull request by tomalin**
1) project/settings_hls.tcl provides recommended HLS compilation settings
2) project/script_*.tcl all changed, so all now have common style and all run settings_hls.tcl
3) IntegrationTests/compile_hdl.tcl does Vivado synthesis & implementation using compilation settings that optimise high frequency performance.

### CODE REVIEW

File: project/script_IR.tcl [hunk 1]
```diff
@@ -1,20 +1,31 @@
...
+add_files -tb ../emData/IL/
+
+# solutions
+open_solution "solution1"
+source settings_hls.tcl
 source set_fpga.tcl
-create_clock -period 4 -name default
-#source "./projrouter/solution1/directives.tcl"
-csim_design -compiler gcc
+create_clock -period 250MHz -name default
```
Comment (pwittich) [inline project/script_IR.tcl]: Shall we put the clock frequency into one of the global TCL files too to ensure all projects use the same one?

Comment (tomalin) [reply project/script_IR.tcl]: As we explore running at higher frequencies, it is possible that one or two of the projects will successfully run at higher frequencies, before all projects achieve this. That is the only argument I see against setting the clock freq. in the common file. I don't mind either way.

Comment (pwittich) [reply project/script_IR.tcl]: I can imagine two clock speeds at most, maybe we have a common slow clock and a common fast clock?

File: project/script_IR.tcl [hunk 2]
```diff
@@ -1,20 +1,31 @@
...
+
+csim_design -compiler gcc -mflags "-j8"
```
Comment (pwittich) [inline project/script_IR.tcl]: is -j8 passed to make or to the compiler? If the latter then maybe it should be set to something like the output of the `nproc` command.

**Comment by tomalin** [comment]
I moved the setting of the clock frequency (and also of the FPGA type) into settings_hls.tcl, as discussed in HLS meeting, and removed the corresponding lines from the individual algo .tcl scripts.

**Pull request merged**